### ISSUE TITLE
shadow_unrelated triggered instead of shadow_reuse for if-let-bindings

### ISSUE THREAD

**Issue by n47h4n13l** [rust-lang/rust-clippy#8087]
### Summary

For a if-let-binding reusing a variable name, the lint shadow_reuse should be triggered. Instead, I get an error when denying shadow_unrelated.

### Lint Name

shadow_reuse, shadow_unrelated

### Reproducer

I tried this code:

```rust
fn main() {
    false_negative();
    false_positive();
}

#[deny(clippy::shadow_reuse)]
fn false_negative() {
    let answer = Some(42);
    if let Some(answer) = answer {
        println!("The answer is {}", answer);
    }
}

#[deny(clippy::shadow_unrelated)]
fn false_positive() {
    let answer = Some(42);
    if let Some(answer) = answer {
        println!("The answer is {}", answer);
    }
}
```

I expected to see this happen:
An error in false_negative().

Instead, this happened:
An error in false_positive()
